### PR TITLE
ESBJAVA-3688 - Clean up Enrich Mediator

### PR DESCRIPTION
**Pull request by Nirthika**
Clean up Enrich Mediator by removing redundant combination of source and target type
https://wso2.org/jira/browse/ESBJAVA-3688

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java [hunk 1]
```diff
@@ -187,4 +194,35 @@ public QName getTagQName() {
...
+                logger.info("Un-expected target type");
+            }
+            if (targetType == 1) {
+                throw new SynapseException("Envelope not support for target attribute");
+            }
+        }
+
+        // check combination
+        if ((sourceType == 1 && targetType == 0) || (sourceType == 1 && targetType == 2) || (sourceType == 2
+                && targetType == 2)) {
```
Comment (Manuri) [inline modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java]: Can we make the comment more descriptive explaining what this condition really means?

File: modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java [hunk 2]
```diff
@@ -187,4 +194,35 @@ public QName getTagQName() {
...
+            if (sourceType < 0) {
+                logger.info("Un-expected source type");
+            }
+        }
+
+        // target type attribute
+        OMAttribute targetTypeAttr = targetEle.getAttribute(ATT_TYPE);
+        if (targetTypeAttr != null && targetTypeAttr.getAttributeValue() != null) {
+            targetType = convertTypeToInit(targetTypeAttr.getAttributeValue());
+            if (targetType < 0) {
```
Comment (a5anka) [inline modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java]: It is better if we do not use magic numbers in conditions to improve readability. Maybe we can define a constant with value 0.

File: modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java [hunk 3]
```diff
@@ -187,4 +194,35 @@ public QName getTagQName() {
...
+            sourceType = convertTypeToInit(sourceTypeAttr.getAttributeValue());
+            if (sourceType < 0) {
+                logger.info("Un-expected source type");
+            }
+        }
+
+        // target type attribute
+        OMAttribute targetTypeAttr = targetEle.getAttribute(ATT_TYPE);
+        if (targetTypeAttr != null && targetTypeAttr.getAttributeValue() != null) {
+            targetType = convertTypeToInit(targetTypeAttr.getAttributeValue());
```
Comment (a5anka) [inline modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java]: Shouldn't this method name be `convertTypeToInt`?

File: modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java (hunk 1 above)
Comment (a5anka) [inline modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java]: It is better if we do not use magic numbers in conditions to improve readability. Maybe we can define constants for these values.

File: modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java (hunk 2 above)
Comment (Nirthika) [inline modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java]: It was the existing implementation, so need to change in many classes. Clear comments are added to explaining what this condition really means.

**Pull request merged**